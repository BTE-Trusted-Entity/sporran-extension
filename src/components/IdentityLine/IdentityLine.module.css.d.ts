export const __esModule: true;
export const component: string;
export const identity: string;
export const avatar: string;
export const name: string;
