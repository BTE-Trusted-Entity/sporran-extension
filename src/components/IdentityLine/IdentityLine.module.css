.component {
  composes: subline from '../Typography/Typography.module.css';
  margin: 0;
  height: 3.25rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.identity {
  display: flex;
  align-items: center;
  margin: 0;
  justify-content: flex-start;
}

.avatar {
  composes: avatar from '../Avatar/Avatar.module.css';
  height: 1.75rem;
  width: 1.75rem;
  background: none;
  margin-right: 0.75rem;
}

.name {
  width: 18rem;
  text-align: left;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}
