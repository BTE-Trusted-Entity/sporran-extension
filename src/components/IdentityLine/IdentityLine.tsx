import * as styles from './IdentityLine.module.css';

import { Identity } from '../../utilities/identities/types';
import { Avatar } from '../Avatar/Avatar';
import { KiltAmount } from '../KiltAmount/KiltAmount';
import { useAddressBalance } from '../Balance/Balance';

interface Props {
  identity: Identity;
  className?: string;
}

export function IdentityLine({
  identity,
  className = styles.component,
}: Props): JSX.Element {
  const balance = useAddressBalance(identity.address);

  return (
    <section className={className}>
      <section className={styles.identity}>
        <Avatar identity={identity} className={styles.avatar} />
        <p className={styles.name}>{identity.name}</p>
      </section>
      {balance && <KiltAmount amount={balance.total} type="funds" />}
    </section>
  );
}
