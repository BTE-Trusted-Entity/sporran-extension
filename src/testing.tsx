export * from '@testing-library/react';
export { waitForElementToBeRemoved } from '@testing-library/dom';
import { act, render as externalRender, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import dialogPolyfill from 'dialog-polyfill';

import { AccountsProviderMock } from './testing/AccountsProviderMock';

export {
  AccountsProviderMock,
  accountsMock,
} from './testing/AccountsProviderMock';

export { mockBackgroundScript } from './testing/mockBackgroundScript';

export function render(
  ui: Parameters<typeof externalRender>[0],
  options?: Parameters<typeof externalRender>[1],
): ReturnType<typeof externalRender> {
  return externalRender(
    <AccountsProviderMock>
      <MemoryRouter>{ui}</MemoryRouter>
    </AccountsProviderMock>,
    options,
  );
}

export async function waitForTooltipUpdate(): Promise<void> {
  expect(await screen.findByRole('status')).toHaveAttribute(
    'data-popper-escaped',
  );
}

const dialogPromise = Promise.resolve();
(dialogPolyfill.registerDialog as jest.Mock).mockReturnValue(dialogPromise);

export async function waitForDialogUpdate(): Promise<void> {
  await act(() => dialogPromise);
}
