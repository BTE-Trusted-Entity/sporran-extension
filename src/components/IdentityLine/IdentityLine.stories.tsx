import { Meta } from '@storybook/react';

import { identitiesMock } from '../../utilities/identities/IdentitiesProvider.mock';

import { IdentityLine } from './IdentityLine';

export default {
  title: 'Components/IdentityLine',
  component: IdentityLine,
} as Meta;

const identity =
  identitiesMock['4pUVoTJ69JMuapNducHJPU68nGkQXB7R9xAWY9dmvUh42653'];

export function Template(): JSX.Element {
  return <IdentityLine identity={identity} />;
}
