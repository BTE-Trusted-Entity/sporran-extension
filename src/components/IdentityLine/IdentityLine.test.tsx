import { identitiesMock, render } from '../../testing/testing';

import { IdentityLine } from './IdentityLine';

const identity =
  identitiesMock['4tDjyLy2gESkLzvaLnpbn7N61VgnwAhqnTHsPPFAwaZjGwP1'];

describe('IdentityLine', () => {
  it('should match the snapshot', async () => {
    const { container } = render(<IdentityLine identity={identity} />);
    expect(container).toMatchSnapshot();
  });
});
