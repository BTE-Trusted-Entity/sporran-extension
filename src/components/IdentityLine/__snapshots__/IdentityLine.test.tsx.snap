// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdentityLine should match the snapshot 1`] = `
<div>
  <section
    class="component"
  >
    <section
      class="identity"
    >
      <div
        class="avatar"
      >
        Identicon
      </div>
      <p
        class="name"
      >
        Light DID Identity
      </p>
    </section>
    <span
      class="amount"
    >
      1.2340
       
      <span
        aria-label="KILT Coin"
        class="coin"
      />
    </span>
  </section>
</div>
`;
